part2_client.py

```diff
@@ -1,58 +1,48 @@
+        self.serverName = serverName
+        self.serverPort = serverPort
+        self.bufferSize = bufferSize
+        self.clientSocket = socket(AF_INET, SOCK_DGRAM)
+        self.clientSocket.settimeout(1.0) # timeout exception will be raised 1 second after trying to receive data
+    
+    def close(self):
+        self.clientSocket.close()
+
+    def ping(self, request: str):
+        try: # use a try/except to handle a timeout from the server side
+            tripStartTime = time.time() # save the time of right before the request is made
+            self.clientSocket.sendto(message.encode(), (self.serverName, self.serverPort)) # send request to server
+            response, serverAddress = self.clientSocket.recvfrom(self.bufferSize) # fetch the response
+            roundTripTime = round(1000 * (time.time() - tripStartTime), 3) # calculate the round-trip-time in ms
+            print(f"Request:\t{request}\nResponse:\t{response.decode()}\nRound trip:\t{roundTripTime}ms")
+        except timeout:
+            print("Request timed out")
+        except Exception:
+            print("Unknown exception occurred")
 
-from socket import *
-import time
 
 if __name__ == "__main__":
     # Pre-define a list of messages to ping the server with
-    messageList = [
+    messages = [
         "PING 1 - hello world!",
         "PING 2 - what is your name?",
-        "PING 3 - SPAM12345",
+        "PING 3 -    SPAM12345",
         "PING 4 - 99999900009999999",
         "PING 5 - ______^^^^^__===------         ___"
     ]
 
-    # Create variables to idenfity the server
-    serverName = "localhost"
-    serverPortParams = ("127.0.0.1", 12000)
-    clientSocket = socket(AF_INET, SOCK_DGRAM)
-
-    # Set a timeout for 1 second while client waits for serve
-    clientSocket.settimeout(1.0)
-
-    # Ping the server once with each message
-    for message in messageList:
-        
-        # Start RTT time tracker
-        messageTimerStart = time.time()
-
-        # Send message over port
-        clientSocket.sendto(message.encode(), serverPortParams)
-
-        # Use a try/except to handle a timeout from the server side
-        try:
-            # Read reply
-            modifiedMessage, serverAddress = clientSocket.recvfrom(2048)
-        
-            # End RTT time tracker
-            messageTimerEnd = time.time()
-
-            # Calculate the RTT
-            roundTripTime = messageTimerEnd - messageTimerStart
-
-            # Print reply and RTT for the message
-            print(f"The recieved message: {modifiedMessage.decode()}\nThe RTT for the message: {roundTripTime} Seconds")
-
-        except timeout:
-            # If no reply is avaliable print the timeout message
-            print("request timed out")
-
-    # Close client after each message is sent
-    clientSocket.close()
\ No newline at end of file
+    client = Client()
+    for message in messages:
+        client.ping(message)
+        print("----")
```

Comment:
I don't think the specs required any additional spacing, but nothing bad. Not sure how picky they are with the following specs.

Reply:
this can also be commented out! I'm paraphrasing but the handout says the client should "print out the RTT and the modified message"